### DISCUSSION TITLE
Does the quadcover working for 3D solid meshes

### DISCUSSION THREAD

**Discussion by zcwang6** [libigl/libigl#1887] [Q&A]
Hello libigl team, thx for providing such a good mesh package. currently, I am using example 505 to solve my 2D singularity problems on my vector fields which work perfectly;  however, it seems only 3D surface meshes working in this algorithm and I guess it must be some reason for that. By any chance, if this algorithm can work for 3D solid meshes to do cubic meshing? Thanks for your time.

BEST

**Comment by danielepanozzo** [comment]
We do not support hexahedral meshing in libigl, the algorithms for volumetric meshing are very different than those for surfaces.

**Reply by zcwang6** [reply]
Thx, got it.
